proof_range.go

```diff
@@ -20,8 +21,21 @@ type RangeProof struct {
 	treeEnd int // 0 if not set, 1 if true, -1 if false.
 }
 
+func (proof *RangeProof) Keys() (keys [][]byte) {
```

Comment:
This is only used in the test below so far. Do you assume this will be useful for other purposes, too? As far as I can see, the keys are always returned (now) together with the range-proof.

Reply:
I think so yeah, the keys returned are the queried keys.  the .Keys() in a RangeProof are all the keys for values queried + any other keys like the item before startKey if startKey is provided (not nil) and doesn't exist in the iavl store, so they're different in functionality.  I'll add a comment saying so.

Reply:
```go
// Keys returns all the keys in the RangeProof.  NOTE: The keys here may
// include more keys than provided by tree.GetRangeWithProof or
// VersionedTree.GetVersionedRangeWithProof.  The keys returned there are only
// in the provided [startKey,endKey){limit} range.  The keys returned here may
// include extra keys, such as:
// - the key before startKey if startKey is provided and doesn't exist;
// - the key after a queried key with tree.GetWithProof, when the key is absent.
func (proof *RangeProof) Keys() (keys [][]byte) {

```

Reply:
Thanks for the explanation. That makes sense.